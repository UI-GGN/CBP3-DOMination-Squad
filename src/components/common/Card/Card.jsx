import { Container, Content } from './Card.style';

const Card = ({ children }) => {
  return (
    <Container>
      <Content>{children}</Content>
    </Container>
  );
};

export default Card;
