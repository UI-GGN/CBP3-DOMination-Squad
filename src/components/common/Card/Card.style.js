import styled from 'styled-components';

export const Content = styled.div`
  padding: 16px 20px;
`;

export const Container = styled.div`
  padding: 16px 20px;
  background-color: ${(props) => (props.color ? props.color : 'white')};
  flex-shrink: 0;
  width: ${(props) => (props.width ? props.width : '325px')};
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  margin: 20px;
  &:hover {
    box-shadow: 0 10px 16px 0 rgba(0, 0, 0, 0.2);
  }
`;
