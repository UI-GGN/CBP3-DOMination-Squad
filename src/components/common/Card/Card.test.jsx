import Card from './Card';
import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

describe('Card', () => {
  it('should render Card component', () => {
    const { container } = render(<Card />);

    expect(container).toMatchSnapshot();
  });
});
