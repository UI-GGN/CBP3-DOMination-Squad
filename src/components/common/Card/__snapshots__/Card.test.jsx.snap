// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > should render Card component 1`] = `
<div>
  <div
    class="sc-gueYoa grPacT"
  />
</div>
`;
